app/assets/javascripts/components/conditions_form_field.jsx

```diff
@@ -0,0 +1,60 @@
+class ConditionsFormField extends React.Component {
+
+  constructor(props) {
+    super();
+    this.getFieldData = this.getFieldData.bind(this)
+    this.updateFieldData = this.updateFieldData.bind(this)
+    this.formatRefQingOptions = this.formatRefQingOptions.bind(this)
+    this.buildUrl = this.buildUrl.bind(this)
+    this.state = props;
+  }
+
+  updateFieldData(ref_qing_id) {
+
+    this.getFieldData(ref_qing_id);
+  }
+
+  getFieldData(ref_qing_id) {
+    ELMO.app.loading(true)
+    var self = this;
+    var url = this.buildUrl(ref_qing_id)
+    $.ajax(url)
+      .done(function(response) {
+          self.setState(response);
+        })
```

Comment:
Should I do more to check that the response is valid state here, or wait for an error in render? As in, if there's a 200 response but not the right structure for json?